### PR TITLE
add real-space RMG converter to convert4qmc

### PR DESCRIPTION
**Pull request by kgasperich**
## Proposed changes

This PR adds RMG hdf5 output to the list of formats supported by convert4qmc (this hdf5 output is generated when the RMG input contains `write_qmcpack_restart = "true"`).

Handling of pseudopotential files is not perfect, so this should be cleaned up in a later PR (the PP terms in the Hamiltonian in the generated `*.qmc.in-wf{j,noj}.xml` will always reference `X.qmcpp.xml` where `X` is the appropriate element symbol, but `convert4qmc` will not generate/link/move/copy/convert these from the PP files used by RMG).

## What type(s) of changes does this code introduce?

- New feature

### Does this introduce a breaking change?

- No

## What systems has this change been tested on?
 cooley (RHEL 7; x86_64) icc 19.1.0.166
## Checklist

_Update the following with a yes where the items apply. If you're unsure about any of them, don't hesitate to ask.  This is
simply a reminder of what we are going to look for before merging your code._

- Yes. This PR is up to date with current the current state of 'develop'
- Yes. Code added or changed in the PR has been clang-formatted
- Yes*. This PR adds tests to cover any new code, or to catch a bug that is being fixed
- Yes. Documentation has been added (if appropriate)

### CODE REVIEW

**Comment by PDoakORNL** [comment]
any tests at all?

**Comment by kgasperich** [comment]
I can add one (or more) to https://github.com/QMCPACK/qmcpack/tree/develop/tests/converter
The smallest ones I have right now (all spin-unpolarized) are 5, 80, 95, and 136 MB (size of the hdf5 output from RMG), and I can add any subset of those.
In terms of coverage, it might be good to have a spin-polarized system, but the smallest of those that I have right now is 1.3 GB; I can try to converge that on a coarser grid to get the size down.

**Comment by PDoakORNL** [comment]
#3354 is relevant to this.  We probably don't want to hold the PR for that issue to be resolved.

**Comment by kgasperich** [comment]
Ok, I added two tests (2.5 and 5 MB), one spin-polarized and one unpolarized.
We can move things around as necessary depending on how #3354 is resolved

**Review by prckent — COMMENTED** [review]
Thanks Kevin.

Questions on the test choices:

what does the k-point grid in the diamond test case achieve? can we save some filespace by using 1x1x1?

for the o atom case, can we make it even more distinct? it is spin polarized but can we readily support open boundary conditions? (relatively unique feature of rmg, hits more code paths)

**Comment by prckent** [comment]
Start testing in-house

**Comment by prckent** [comment]
Test this please

**Comment by kgasperich** [comment]
@prckent 
> what does the k-point grid in the diamond test case achieve? can we save some filespace by using 1x1x1?

In the interest of expediency I was just selecting a subset of the existing RMG example tests; I wanted to use the O atom for a spin-polarized example, and because that was 1x1x1 I thought it might provide better coverage to use a grid with more than 1 cell for the spin-unpolarized test (diamond).  The diamond test uses a 4x4x4 grid, but I can change that to 1x1x1 if the grid isn't important (or to 1x1x2 or 2x2x2 if we want something smaller than what we have, but larger than 1x1x1).

> for the o atom case, can we make it even more distinct? it is spin polarized but can we readily support open boundary conditions? (relatively unique feature of rmg, hits more code paths)

I'm not sure I understand the question; I was under the impression that RMG only supported periodic boundary conditions.
https://github.com/RMGDFT/rmgdft/blob/606ae210d50b1bf76e228ff98f1a9058e560c499/Input/ReadCommon.cpp#L423-L426

**Comment by prckent** [comment]
Use the smallest grid for file size. Forget my comments on non-periodic settings.

**Comment by kgasperich** [comment]
@prckent I added some ctest framework and a working example for RMG.  I still need to figure out how/where to set the path to the RMG executable (maybe best to just add a cmake variable `RMG_BIN` analogous to `QE_BIN`) and I'll need to add some logic so that the RMG tests are added only if RMG is found (something similar to [FindQE.cmake](https://github.com/QMCPACK/qmcpack/blob/develop/CMake/FindQE.cmake)).

**Comment by kgasperich** [comment]
@prckent the RMG testing should be working now as long as `RMG_BIN` is passed to cmake.  Should we add an RMG build step to the nightly test scripts?

**Comment by prckent** [comment]
Is it safe to use v4.2.2 of RMG or do we need to use develop?

**Comment by prckent** [comment]
Working
```
pk7@sulfur build_quick]$ ctest -R rmg --output-on-failure
Test project /home/pk7/projects/qmc/git_QMCPACK_prckent/build_quick
      Start   62: converter_test_diamond2_rmg
 1/15 Test   #62: converter_test_diamond2_rmg ............................   Passed    0.39 sec
      Start   63: converter_test_O_rmg
 2/15 Test   #63: converter_test_O_rmg ...................................   Passed    0.36 sec
      Start  228: rmg-Diamond2-1x1x1-gamma-ccECP-np-1-scf
 3/15 Test  #228: rmg-Diamond2-1x1x1-gamma-ccECP-np-1-scf ................   Passed    6.70 sec
      Start  229: LINK_rmg-Diamond2-1x1x1-gamma-ccECP-np-1_h5_Waves
 4/15 Test  #229: LINK_rmg-Diamond2-1x1x1-gamma-ccECP-np-1_h5_Waves ......   Passed    0.05 sec
      Start  230: rmg-Diamond2-1x1x1-gamma-ccECP-np-1-rmg2qmc
 5/15 Test  #230: rmg-Diamond2-1x1x1-gamma-ccECP-np-1-rmg2qmc ............   Passed    0.29 sec
      Start  231: LINK_rmg-Diamond2-1x1x1-gamma-ccECP-np-1_TO_1-16
 6/15 Test  #231: LINK_rmg-Diamond2-1x1x1-gamma-ccECP-np-1_TO_1-16 .......   Passed    0.05 sec
      Start  232: COPY_rmg-Diamond2-1x1x1-gamma-ccECP-np-1_XML_TO_1-16
 7/15 Test  #232: COPY_rmg-Diamond2-1x1x1-gamma-ccECP-np-1_XML_TO_1-16 ...   Passed    0.08 sec
      Start  233: rmg-Diamond2-1x1x1-gamma-ccECP-np-1-1-16
 8/15 Test  #233: rmg-Diamond2-1x1x1-gamma-ccECP-np-1-1-16 ...............   Passed    7.54 sec
      Start  234: rmg-Diamond2-1x1x1-gamma-ccECP-np-1-1-16-kinetic
 9/15 Test  #234: rmg-Diamond2-1x1x1-gamma-ccECP-np-1-1-16-kinetic .......   Passed    0.07 sec
      Start  235: rmg-Diamond2-1x1x1-gamma-ccECP-np-1-1-16-samples
10/15 Test  #235: rmg-Diamond2-1x1x1-gamma-ccECP-np-1-1-16-samples .......   Passed    0.07 sec
      Start  236: rmg-Diamond2-1x1x1-gamma-ccECP-np-1-1-16-localecp
11/15 Test  #236: rmg-Diamond2-1x1x1-gamma-ccECP-np-1-1-16-localecp ......   Passed    0.07 sec
      Start  237: rmg-Diamond2-1x1x1-gamma-ccECP-np-1-1-16-nonlocalecp
12/15 Test  #237: rmg-Diamond2-1x1x1-gamma-ccECP-np-1-1-16-nonlocalecp ...   Passed    0.07 sec
      Start 2034: ntest_nexus_rmg_input
13/15 Test #2034: ntest_nexus_rmg_input ..................................   Passed    0.71 sec
      Start 2043: ntest_nexus_rmg_analyzer
14/15 Test #2043: ntest_nexus_rmg_analyzer ...............................   Passed    0.58 sec
      Start 2051: ntest_nexus_rmg_simulation
15/15 Test #2051: ntest_nexus_rmg_simulation .............................   Passed    0.56 sec

100% tests passed, 0 tests failed out of 15
```

**Comment by prckent** [comment]
@kgasperich What are your plans to update this? I would really like to get it merged and in use.  Additional updates can be done in later PRs.

**Comment by kgasperich** [comment]
@prckent sorry, it fell off my radar after the last meeting, but I've updated it now and brought it up-to-date with develop

File: CMake/FindRMG.cmake [hunk 1]
```diff
@@ -0,0 +1,12 @@
+# Locate rmg-cpu 
+# Take RMG_BIN as hint for location
+
+find_path(RMG_CPU_DIR rmg-cpu HINTS ${RMG_BIN})
```
[the remark this answers is not in the document]
Comment (kgasperich) [reply CMake/FindRMG.cmake]: that does look cleaner; I've changed it

File: src/QMCTools/RMGParser.h [hunk 2]
```diff
@@ -0,0 +1,40 @@
+//////////////////////////////////////////////////////////////////////////////////////
+// This file is distributed under the University of Illinois/NCSA Open Source License.
+// See LICENSE file in top directory for details.
+//
+// Copyright (c) 2016 Jeongnim Kim and QMCPACK developers.
```
[the remark this answers is not in the document]
Comment (kgasperich) [reply src/QMCTools/RMGParser.h]: fixed

File: src/QMCTools/RMGParser.h [hunk 3]
```diff
@@ -0,0 +1,40 @@
...
+
+class RMGParser : public QMCGaussianParserBase, public OhmmsAsciiParser
+{
+public:
+  RMGParser();
+
+  RMGParser(int argc, char** argv);
+
+  std::vector<std::string> ECP_names;
+  //std::vector<std::string> tags;
```
[the remark this answers is not in the document]
Comment (kgasperich) [reply src/QMCTools/RMGParser.h]: done

**Review by prckent — APPROVED** [review]

**Comment by prckent** [comment]
Start testing in-house

**Comment by ye-luo** [comment]
Test this please

**Comment by ye-luo** [comment]
I saw binary h5 files are added but we also have ways to generate them. Can we remove those h5 and bake converter test into the workflow test? If we are in favor of merging this as it is. Need to squash to mare sure no intermediate files stored in the repo. When these h5 files need updating, they should just be removed.

**Comment by prckent** [comment]
I think this is consistent with our QE tests. We can't test the converter without pregenerated files in the repo.
